ykllvmwrap/src/ykllvmwrap.cc

```diff
@@ -397,6 +397,11 @@ void *compileIRTrace(FN Func, char *FuncNames[], size_t BBs[], size_t TraceLen,
   std::tie(JITMod, TraceName, GlobalMappings, AOTMappingVec) =
       Func(AOTMod, FuncNames, BBs, TraceLen, FAddrKeys, FAddrVals, FAddrLen);
 
+  // If we failed to build the trace, return null.
+  if (JITMod == nullptr) {
+    return nullptr;
```

Comment:
There are going to be a few different cases where we abort trace compilation. Would it be an idea to build in (now or soon) some way of indicating *why* we couldn't build the trace?

In the test we have:
```
jit-state: invalid trace
```

I think that should eventually be something more like:
```
jit-state: abort-trace-compilation (trace uses setjmp/longjmp)
```

This could be a separate PR. It will require re-engineering of the return value (possibly using pointer tagging or a union?).

Reply:
It would be great to have better "errors". But funneling errors from `jitmodbuilder` via FFI back to Rust sounded a bit more involved than what I wanted to achieve with this PR.

Reply:
Yeah. It's a bit annoying because you would need a way to express the error inside the pointer return value.

Reply:
Raise an issue and link in FIXME?

Reply:
[d6951d5](https://github.com/ykjit/yk/pull/609/commits/d6951d5e61db84d131ee1f5f4c7c2222509d2c8f)

Reply:
Good idea. I'm not really sure *what* we should do here: it probably has to be quite lightweight. Punting for now is fine.